File: compiler-group-traslator/src/test/resources/c/IntShiftRight.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;
int shiftRight(int i, int j) {
	return i >> j;

	//set all vars to zero and have the TEMP variable for the return expression
}
int main(char* args) {
	vm = newVM();
	assertTrue(2 >> 1 == 1);
	assertTrue(-2 >> 1 == -1);
	assertTrue(shiftRight(2, 1) == 1);
	assertTrue(shiftRight(-2, 1) == -1);
	return 0;
}

File: compiler-solo-ast/src/test/resources/IntShiftRight.java
public class IntShiftRight {
    public static void main(String[] args) {
        StaticJavaLib.assertTrue( 2 >> 1 ==  1);
        StaticJavaLib.assertTrue(-2 >> 1 == -1);

        StaticJavaLib.assertTrue(shiftRight( 2, 1) ==  1);
        StaticJavaLib.assertTrue(shiftRight(-2, 1) == -1);
    }

    static int shiftRight(int i, int j){
        return i >> j;
    }
}
    

